.gitlab/build_lassen.yml

```diff
@@ -15,6 +15,10 @@
 .with_cuda:
   before_script:
     - module load cuda/11.1.1
+    # python3.8 is needed on lassen to avoid trampling on the x86 clingo wheel
+    # This is not the ideal place for it but the pipeline syntax doesn't allow
+    # for multiple before_scripts - see https://gitlab.com/gitlab-org/gitlab-runner/-/issues/2301
```

Comment:
That’s true:
Yaml lists are not merged when used in the `extends` mechanism. True for script, before/after_script, rules, even tags.

An alternate solution would be to have a top level before script (one per machine for example) and adapt the behavior depending on the job name (requires a convention) or maybe better, on variables:
```yaml
.on_lassen:
  before_script:
    - module load python/3.8
    - if [[ $CUDA == "ON" ]]; then module load cuda/11.1.1; fi

.with_cuda:
  variables:
    CUDA: "ON"
```
That’s because variables can be merged through `extends`.